src/wp-includes/class-wp-theme-json.php

```diff
@@ -4531,12 +4533,15 @@ function ( $matches ) use ( $variation_class ) {
 	 * Collects valid block style variations keyed by block type.
 	 *
 	 * @since 6.6.0
+	 * @since 6.8.0 Added the `$blocks_metadata` parameter.
 	 *
+	 * @param array $blocks_metadata Optional list of selectors per block.
 	 * @return array Valid block style variations by block type.
 	 */
-	protected static function get_valid_block_style_variations() {
+	protected static function get_valid_block_style_variations( $blocks_metadata = array() ) {
 		$valid_variations = array();
-		foreach ( self::get_blocks_metadata() as $block_name => $block_meta ) {
```

Comment:
Maybe we should still call `self::get_blocks_metadata()` in case no value was provided, i.e. as default?

Since this method is `protected` (not `private`), in theory someone else could have overwritten this class. Highly unlikely but possible. Additionally though, not _requiring_ the parameter makes using the method more ergonomic in situations where it hasn't been computed before.

Reply:
I already added `$blocks_metadata  = empty( $blocks_metadata ) ? static::get_blocks_metadata() : $blocks_metadata;` if the value is not passed of empty.

Let me know if i missed anything.